### ISSUE TITLE
Problem on ASRock H510 Pro BTC+ motherboard with 5700Xt and W5500  graphics cards on HiveOS

### ISSUE THREAD

**Issue by Serg-19712023** [andru-kun/wildrig-multi#202]
Good day!
I've been struggling with this problem for days, does anyone know the solution?
Of the 6 AMD cards, only 2 work at full capacity. The rest are either idle or work at half strength, depending on the version of firewood. I updated the amd-ocl-install command to 22.10 5.1, two work fine, the rest are idle. On Hive version 5.10.0-hiveos #110
and AMD drivers 21.40.1 - all six cards work, but the performance is 2 times lower.
WildRig version 0.36.2.
5700xt gave 19 Mn, with an updated driver, but only one more worked with it. How to update firewood, and so that all 6 cards work?
Thanks to.
![photo_2023-01-14_19-03-28](https://user-images.githubusercontent.com/122687349/212482639-69674a40-4319-4b9a-a38d-86169f9696db.jpg)
![photo_2023-01-14_19-03-59](https://user-images.githubusercontent.com/122687349/212482641-eab747be-3d53-4d12-93ef-4084c0580769.jpg)

**Comment by andru-kun** [comment]
Have you installed the latest stable HiveOS first? Using command hive-replace -s
You need to do so before you install new drivers with amd-ocl-install command.

**Comment by Serg-19712023** [comment]
After entering the hive-replace -s command and updating to a stable version, the OS gives an error when running amd-ocl-install 22.10 5.1 - "not found command"

**Comment by andru-kun** [comment]
> After entering the hive-replace -s command and updating to a stable version, the OS gives an error when running amd-ocl-install 22.10 5.1 - "not found command"

Update HiveOS via Upgrade button(you do that when you want new versions of miners).

**Comment by Serg-19712023** [comment]
That's what I did after this error.  After that, the driver update occurs
without error, but, as a result, only 2 video cards are mined at full
strength, the rest are idle.

вс, 15 янв. 2023 г., 0:21 Andrii ***@***.***>:

**Comment by andru-kun** [comment]
Does miner see them at all? Can you check miner console?

**Comment by andru-kun** [comment]
Also try driver 22.20 instead of 22.10

**Comment by Serg-19712023** [comment]
Yes, the miner, after updating the driver, sees 2 video cards.  Like in the screenshot.

**Comment by Serg-19712023** [comment]
22.20 tried.  The result is the same.

**Comment by andru-kun** [comment]
Well, I have no more ideas

**Comment by Serg-19712023** [comment]
🤔😔😔😔

**Comment by Serg-19712023** [comment]
I'll try another recipe tomorrow.  Then I will write about the results.

**Comment by Serg-19712023** [comment]
On this rig I was mining ETH, ETC, Aleo, Kaspa, all cards had the same performance, and all cards were visible.  And here is such a thing.

**Comment by Serg-19712023** [comment]
Доброе утро, Андрей!
Только сейчас дошло, что можно не мучать гугл-переводчик:-)
В общем, переключил PCI в режим Gen3 (было Gen1) и хэшрейт добавился +20%.
Обновил дрова до версии 22.20.5 - опять отвал 4-х карт, при повышенном хэшрейте у работающих двух.
Пришлось опять накатить стабильную версию, и обновить ее до текущей.
Так что результат вот такой. 
В принципе, и прибавка 20% конечно радует. Но, есть куда расти, та же 5700Хт может легко давать 19.65 Мн, вместо текущих 15.44Мн. И W5500 может 10.6 вместо 8.4Мн.
Успехов в разработке!
Слава Украине!

![photo_2023-01-15_08-14-43](https://user-images.githubusercontent.com/122687349/212524318-2bc4479f-ea16-4aba-ade9-8762f9d34e09.jpg)
![photo_2023-01-15_08-15-02](https://user-images.githubusercontent.com/122687349/212524320-c6d7106c-3a06-43d5-aeaa-0b78cebd78ba.jpg)
![photo_2023-01-15_08-15-10](https://user-images.githubusercontent.com/122687349/212524321-fedb84b2-2df8-4b51-b87c-6f7f6faad50f.jpg)

**Comment by Serg-19712023** [comment]
Можно еще 14 Мн выжать без напряга, осталось дождаться нормальной связки ядра+дрова :-)

**Comment by andru-kun** [comment]
> Доброе утро, Андрей!
Доброе :) Увы, очень странная ситуация, такое ощущение, что беда именно в поддержке W5500 на новых дровах, но самый странный момент тут - что одна карта из этой серии инициализируется, а остальные нет. А что если из рига вынять 5700XT на новых дровах?
